### DISCUSSION TITLE
How to select only first n elements?

### DISCUSSION THREAD

**Discussion by art-qa** [cypress-io/cypress#22362] [Questions and Help]
I have a table with a list of messages that are filtered. There is a method each to go through every each of them. But I want to go through only first 3 (divs) of them. How can it be done? The classes are all the same.

```
<div class="messages">
  <div class="messages_messagesList">
    <div>
	<div class="inbox-message">text1</div>
	<div class="inbox-message">text1</div>
	<div class="inbox-message">text1</div>
	<div class="inbox-message">text1</div>
	<div class="inbox-message">text1</div>
	<div class="inbox-message">text1</div>
    </div>
  </div>
</div>
```

**Comment by jjhelguero** [comment]
You can get an array of all the messages that are filtered by `.inbox-message` elements and with that array you can use the jquery extension selector [`:lt()`](https://api.jquery.com/lt-selector/), which select all elements at an index  less than what is supplied.

Here is a working [example](https://github.com/jjhelguero/cypress-examples/blob/main/cypress/integration/getNElements.md).

```js
cy.get(".inbox-message") // returns array of all matching elements
   .filter(":lt(3)") // filter all matching elements
   .should('have.length', 3)
```

**Reply by josemanuelep** [reply]
Nice!
